mitiq/observable/pauli.py

```diff
@@ -282,8 +282,27 @@ def _measure_in(
 
         basis_rotations = set()
         support = set()
+
+        # Find any existing measurement gates in the circuit
+        existing_measurements = []
+        measurement_tuples = list(
+            circuit.findall_operations_with_gate_type(cirq.MeasurementGate)
+        )
+        if measurement_tuples:
+            existing_measurements = [
+                measurement_tuple[1].qubits[0]
+                for measurement_tuple in measurement_tuples
+            ]
```

Comment:
Two suggestions:

1. I'm not sure this code will work if the circuit has a "2d" measurement[^1]. We can instead get all the qubits used in measurement if we have a set that keeps track of the qubits themselves
    ```py
    qubits_with_measurements = set()
    for _, op, _ in c.findall_operations_with_gate_type(cirq.MeasurementGate):
        qubits_with_measurements.update(op.qubits)
    ```

2. Once we have `qubits_with_measurements`, we can make the check to raise the error a little simpler:
    ```py
    if support & qubits_with_measurements:
        raise ValueError(...)
    ```


[^1]: E.g. if the circuit has `cirq.measure(LineQubit(1), LineQubit(2))`.

Reply:
As a note for historical purposes, https://github.com/unitaryfund/mitiq/pull/2514/commits/a0e4cda6a2d76e7dbff43e111cfa0371df123fbe changed the test to include a "2d" measurement to ensure we cover that use case.